File: redsheet-portal/src/app/database-models/long-term-objective.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ProjectBackground,  vLongTermObjectiveRecord  } from './'
export class LongTermObjective {
    constructor(longTermObjective?: LongTermObjective  | vLongTermObjectiveRecord )  {
        let properties = [ "longTermObjectiveId", "longTermObjectiveValue", "projectBackgroundId",];
        for (let property of properties)
            if (longTermObjective && longTermObjective[property])
                this[property] = longTermObjective[property];
    }
 
	longTermObjectiveId : number =   0 ;
 
    @required()
    @maxLength(100)
	longTermObjectiveValue : string =   undefined;
 
    @range(0,2147483647)
	projectBackgroundId : number =   undefined;
	projectBackground : ProjectBackground  ;


}

File: RedSheet.DbEntities/Models/LongTermObjective.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("LongTermObjectives", Schema = "dbo")]
	[Serializable]
	public partial class LongTermObjective  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int LongTermObjectiveId { get; set; } 
 
	[Required] 
	[MaxLength(100)]
	public string LongTermObjectiveValue { get; set; } 
 
	[RelationshipTableAttribue("ProjectBackgrounds","dbo")]
	//Relationships
	public int ProjectBackgroundId { get; set; } 
 
	[ForeignKey("ProjectBackgroundId")]
	public ProjectBackground ProjectBackground { get; set; } 

	public LongTermObjective()
	{
    }

    }
}

